internal/repository/contact.go

```diff
@@ -10,7 +11,7 @@ type ContactRepository interface {
 	GetByID(id uint) (model.Contact, error)
 	GetByUserID(id uint) ([]model.Contact, error)
 	Update(contact model.Contact) (model.Contact, error)
-	DeleteByID(id uint) error
+	DeleteByID(userID, id uint) error
```

Comment:
Method name does not contains `userID`. It should be named `DeleteByUserIDAndID`